### PR TITLE
Add Knip

### PR DESCRIPTION
**Pull request by manuelblum**
Add [Knip](https://knip.dev/) to lint pipelines. Knip can support in linting dead source code, unused files, unused dependencies, and unnecessary exports.

### CODE REVIEW

File: demo/admin/codegen.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
+import { resolve } from "node:path";
+
 import { CodegenConfig } from "@graphql-codegen/cli";
 import { readFileSync } from "fs";
 import { buildSchema } from "graphql";
 
-const schema = buildSchema(readFileSync("./schema.gql").toString());
-
+const schema = buildSchema(readFileSync(resolve(__dirname, "./schema.gql")).toString());
```
Comment (johnnyomair) [inline demo/admin/codegen.ts:7]: Why is this necessary? 🤔

File: knip.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            "project": ["./src/**/*.{ts,tsx}"]
+        },
+        "demo/admin": {
+            "entry": ["./src/loader.ts", "./src/**/*.cometGen.ts"],
+            "project": ["./src/**/*.{ts,tsx}", "!./src/**/generated/**"],
+            "ignoreDependencies": ["@swc/plugin-emotion"]
+        }
+    },
+    "rules": {
+        "enumMembers": "off"
```
Comment (johnnyomair) [inline knip.json:70]: If we turn it off here, we probably should turn it off in Starter as well.

File: package.json [hunk 3]
```diff
@@ -40,13 +41,14 @@
...
         "@types/node": "^22.0.0",
+        "knip": "^5.38.1",
         "cspell": "^8.15.4",
         "dotenv-cli": "^7.0.0",
         "husky": "^7.0.4",
         "lint-staged": "^12.3.7",
         "prettier": "^2.0.0",
         "rimraf": "^3.0.2",
-        "typescript": "^4.2.2",
+        "typescript": "^5.7.2",
```
Comment (johnnyomair) [inline package.json:51]: Why do you need to upgrade TS? I believe this was tried before and caused some problems.

File: packages/admin/admin-date-time/src/dateTimePicker/DateTimePicker.tsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ import { TimePicker as TimePickerBase } from "../timePicker/TimePicker";
 import { getIsoDateString } from "../utils/datePickerHelpers";
 import { getDateWithNewTime, getTimeStringFromDate } from "../utils/timePickerHelpers";
 
-export type DateTimePickerClassKey = "root" | "dateFormControl" | "timeFormControl" | "datePicker" | "timePicker";
+type DateTimePickerClassKey = "root" | "dateFormControl" | "timeFormControl" | "datePicker" | "timePicker";
```
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/dateTimePicker/DateTimePicker.tsx:12]: @jamesricky is it okay to remove this export? Does this cause problems with theming?

File: packages/admin/admin-date-time/src/timePicker/FinalFormTimePicker.tsx [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { FieldRenderProps } from "react-final-form";
 
 import { TimePicker, TimePickerProps } from "./TimePicker";
 
-export type FinalFormTimePickerProps = TimePickerProps & FieldRenderProps<string, HTMLInputElement | HTMLTextAreaElement>;
+type FinalFormTimePickerProps = TimePickerProps & FieldRenderProps<string, HTMLInputElement | HTMLTextAreaElement>;
```
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/timePicker/FinalFormTimePicker.tsx:5]: @jamesricky is it okay to remove this export? Does this cause problems with theming?

File: packages/admin/admin/src/index.ts [hunk 6]
```diff
@@ -26,6 +26,7 @@ export { FieldSet, FieldSetClassKey, FieldSetProps } from "./common/FieldSet";
 export { HoverActions, HoverActionsClassKey, HoverActionsProps } from "./common/HoverActions";
 export { Loading, LoadingProps } from "./common/Loading";
 export { ToolbarActions, ToolbarActionsClassKey } from "./common/toolbar/actions/ToolbarActions";
+export { ToolbarActionButton } from "./common/toolbar/actions/ToolbarActionsButton";
```
Comment (johnnyomair) [inline packages/admin/admin/src/index.ts:29]: Do we need a changeset for this?

File: packages/admin/cms-admin/src/react-split.d.ts [hunk 7]
```diff
@@ -1,54 +0,0 @@
-declare module "react-split" {
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/react-split.d.ts:1]: Is it okay to remove this? Don't we need the types for react-split?

File: packages/eslint-plugin/package.json [hunk 8]
```diff
@@ -13,6 +13,14 @@
...
         "test:watch": "jest --watch"
     },
+    "dependencies": {
+        "@typescript-eslint/parser": "^5.0.0",
+        "eslint-config-prettier": "^8.6.0",
+        "eslint-plugin-prettier": "^4.2.1",
+        "eslint-plugin-simple-import-sort": "^9.0.0",
+        "eslint-plugin-unused-imports": "^2.0.0",
+        "@typescript-eslint/eslint-plugin": "^5.48.2"
+    },
```
Comment (johnnyomair) [inline packages/eslint-plugin/package.json:23]: Why does the plugin need the dependencies?

File: packages/api/cms-api/src/index.ts [hunk 9]
```diff
@@ -59,6 +59,7 @@ export { RootBlockEntity } from "./blocks/decorators/root-block-entity";
 export { EmailLinkBlock } from "./blocks/email-link.block";
 export { ExternalLinkBlock } from "./blocks/ExternalLinkBlock";
 export { ColumnsBlockFactory } from "./blocks/factories/columns-block.factory";
+export { BlocksBlockInputInterface } from "./blocks/factories/createBlocksBlock";
```
Comment (johnnyomair) [inline packages/api/cms-api/src/index.ts:62]: Why is this export needed?

File: packages/admin/admin-color-picker/src/ColorPicker.tsx [hunk 10]
```diff
@@ -30,7 +30,7 @@ import {
 } from "./ColorPicker.slots";
 
 export interface ColorPickerColorPreviewProps extends Omit<HTMLAttributes<HTMLDivElement>, "color">, PreviewIndicatorColorProps {}
-export interface ColorPickerNoColorPreviewProps extends HTMLAttributes<HTMLDivElement>, PreviewIndicatorEmptyOrInvalidProps {}
+interface ColorPickerNoColorPreviewProps extends HTMLAttributes<HTMLDivElement>, PreviewIndicatorEmptyOrInvalidProps {}
```
Comment (johnnyomair) [inline packages/admin/admin-color-picker/src/ColorPicker.tsx:33]: This should be added to index.ts

File: packages/admin/admin-date-time/src/DatePickerNavigation.tsx [hunk 11]
```diff
@@ -5,7 +5,7 @@ import { css, Theme, useThemeProps } from "@mui/material/styles";
 import { useRef, useState } from "react";
 import { useIntl } from "react-intl";
 
-export interface DatePickerNavigationProps
+interface DatePickerNavigationProps
```
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/DatePickerNavigation.tsx:8]: This component isn't added to index.ts, so removing should be fine.

File: packages/admin/admin-date-time/src/dateTimePicker/DateTimePicker.tsx (hunk 4 above)
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/dateTimePicker/DateTimePicker.tsx:12]: This should be added to index.ts

File: packages/admin/admin-date-time/src/timePicker/FinalFormTimePicker.tsx (hunk 5 above)
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/timePicker/FinalFormTimePicker.tsx:5]: This should be added to index.ts with deprecation notice

File: packages/admin/admin-date-time/src/timePicker/TimePicker.tsx [hunk 12]
```diff
@@ -15,16 +15,16 @@ import { FormatDateOptions, FormattedTime, useIntl } from "react-intl";
 
 import { getClosestDateToDate, getDateFromTimeValue, getDateRangeListByMinuteStep } from "../utils/timePickerHelpers";
 
-export type TimePickerClassKey = InputWithPopperClassKey | "startAdornment" | "timeOptionsList" | "timeOptionItem";
+type TimePickerClassKey = InputWithPopperClassKey | "startAdornment" | "timeOptionsList" | "timeOptionItem";
```
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/timePicker/TimePicker.tsx:18]: This should be added to index.ts

File: packages/admin/admin-date-time/src/timeRangePicker/TimeRangePicker.tsx [hunk 13]
```diff
@@ -6,14 +6,7 @@ import { FormatDateOptions, FormattedMessage, useIntl } from "react-intl";
...
 
-export type TimeRangePickerClassKey =
-    | "root"
-    | "startFormControl"
-    | "endFormControl"
-    | "timePicker"
-    | "startTimePicker"
-    | "endTimePicker"
-    | "separator";
+type TimeRangePickerClassKey = "root" | "startFormControl" | "endFormControl" | "timePicker" | "startTimePicker" | "endTimePicker" | "separator";
```
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/timeRangePicker/TimeRangePicker.tsx:9]: This should be added to index.ts

File: packages/admin/admin/src/appHeader/fillSpace/AppHeaderFillSpace.tsx [hunk 14]
```diff
@@ -6,7 +6,7 @@ import { ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps
 
 export type AppHeaderFillSpaceClassKey = "root";
 
-export type AppHeaderFillSpaceProps = ThemedComponentBaseProps<{
+type AppHeaderFillSpaceProps = ThemedComponentBaseProps<{
```
Comment (johnnyomair) [inline packages/admin/admin/src/appHeader/fillSpace/AppHeaderFillSpace.tsx:9]: This should be added to index.ts

File: packages/admin/admin/src/common/ClearInputAdornment.tsx [hunk 15]
```diff
@@ -29,7 +29,7 @@ export interface ClearInputAdornmentProps
 
 type OwnerState = Pick<ClearInputAdornmentProps, "position">;
 
-export type ClearInputAdornmentClassKey = InputAdornmentClassKey | "buttonBase";
+type ClearInputAdornmentClassKey = InputAdornmentClassKey | "buttonBase";
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/ClearInputAdornment.tsx:32]: This should be added to index.ts

File: packages/admin/admin/src/dataGrid/CrudMoreActionsMenu.tsx [hunk 16]
```diff
@@ -22,9 +22,9 @@ import { FormattedMessage } from "react-intl";
 import { createComponentSlot } from "../helpers/createComponentSlot";
 import { ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
 
-export type CrudMoreActionsMenuClassKey = "root" | "group" | "divider" | "button" | "chip" | "menuItem";
+type CrudMoreActionsMenuClassKey = "root" | "group" | "divider" | "button" | "chip" | "menuItem";
```
Comment (johnnyomair) [inline packages/admin/admin/src/dataGrid/CrudMoreActionsMenu.tsx:25]: This should be added to index.ts

File: packages/admin/admin/src/table/TableDndOrder.tsx [hunk 17]
```diff
@@ -11,7 +11,7 @@ import { ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
 import { IRow, ITableProps, ITableRowProps, Table, TableColumns, TableHeadColumns } from "./Table";
 import { TableBodyRow } from "./TableBodyRow";
 
-export type TableDndOrderClassKey = "root" | "dragCell" | "dragIconContainer";
+type TableDndOrderClassKey = "root" | "dragCell" | "dragIconContainer";
```
Comment (johnnyomair) [inline packages/admin/admin/src/table/TableDndOrder.tsx:14]: This should be added to index.ts

File: packages/admin/blocks-admin/src/blocks/factories/createBlocksBlock.tsx [hunk 18]
```diff
@@ -69,7 +69,7 @@ export interface BlocksBlockFragment<AdditionalItemFields extends Record<string,
     >;
 }
 
-export interface BlocksBlockOutput<AdditionalItemFields extends Record<string, unknown> = DefaultAdditionalItemFields> {
+interface BlocksBlockOutput<AdditionalItemFields extends Record<string, unknown> = DefaultAdditionalItemFields> {
```
Comment (johnnyomair) [inline packages/admin/blocks-admin/src/blocks/factories/createBlocksBlock.tsx:72]: This should be added to index.ts

File: packages/admin/blocks-admin/src/blocks/factories/createListBlock.tsx [hunk 19]
```diff
@@ -49,7 +49,7 @@ export interface ListBlockFragment<AdditionalItemFields extends Record<string, u
     >;
 }
 
-export interface ListBlockOutput<AdditionalItemFields extends Record<string, unknown> = DefaultAdditionalItemFields> {
+interface ListBlockOutput<AdditionalItemFields extends Record<string, unknown> = DefaultAdditionalItemFields> {
```
Comment (johnnyomair) [inline packages/admin/blocks-admin/src/blocks/factories/createListBlock.tsx:52]: This should be added to index.ts

File: packages/admin/blocks-admin/src/blocks/factories/createOneOfBlock.tsx [hunk 20]
```diff
@@ -49,7 +49,7 @@ export interface OneOfBlockPreviewState extends PreviewStateInterface {
 
 type ActiveType<Config extends boolean> = Config extends false ? { activeType: string } : { activeType?: string };
 
-export type OneOfBlockOutput<Config extends boolean> = {
+type OneOfBlockOutput<Config extends boolean> = {
```
Comment (johnnyomair) [inline packages/admin/blocks-admin/src/blocks/factories/createOneOfBlock.tsx:52]: This should be added to index.ts

File: packages/admin/blocks-admin/src/blocks/factories/createOptionalBlock.tsx [hunk 21]
```diff
@@ -23,7 +23,7 @@ export interface OptionalBlockDecoratorFragment<DecoratedBlock extends BlockInte
     visible: boolean;
 }
 
-export interface OptionalBlockOutput<DecoratedBlock extends BlockInterface> {
+interface OptionalBlockOutput<DecoratedBlock extends BlockInterface> {
```
Comment (johnnyomair) [inline packages/admin/blocks-admin/src/blocks/factories/createOptionalBlock.tsx:26]: This should be added to index.ts

File: packages/admin/cms-admin/src/blocks/PixelImageBlock.tsx [hunk 22]
```diff
@@ -31,8 +31,8 @@ import { useCmsBlockContext } from "./useCmsBlockContext";
 
 export type ImageBlockState = Omit<PixelImageBlockData, "urlTemplate">;
 
-export const urlTemplateRoute = "/dam/images/preview/$fileId/crop:$crop/resize:$resizeWidth:$resizeHeight/$fileName";
-export function createPreviewUrl({ damFile, cropArea }: ImageBlockState, apiUrl: string, resize?: { width: number; height: number }): string {
+const urlTemplateRoute = "/dam/images/preview/$fileId/crop:$crop/resize:$resizeWidth:$resizeHeight/$fileName";
+function createPreviewUrl({ damFile, cropArea }: ImageBlockState, apiUrl: string, resize?: { width: number; height: number }): string {
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/blocks/PixelImageBlock.tsx:35]: This is used in projects, but I'm not sure if this should be exported.

File: packages/admin/cms-admin/src/generator/future/generator.ts [hunk 23]
```diff
@@ -113,7 +113,7 @@ export type FormConfig<T extends { __typename?: string }> = {
     fields: (FormFieldConfig<T> | FormLayoutConfig<T>)[];
 };
 
-export type TabsConfig = { type: "tabs"; tabs: { name: string; content: GeneratorConfig }[] };
+type TabsConfig = { type: "tabs"; tabs: { name: string; content: GeneratorConfig }[] };
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/generator/future/generator.ts:116]: Is this even used at all?

File: packages/site/cms-site/src/iframebridge/withPreview.tsx [hunk 24]
```diff
@@ -12,7 +12,7 @@ export interface WithPreviewProps {
     data: WithPreviewPropsData;
 }
 
-export interface WithPreviewPropsData {
+interface WithPreviewPropsData {
```
Comment (johnnyomair) [inline packages/site/cms-site/src/iframebridge/withPreview.tsx:15]: This is used in projects, but shouldn't be used. Devs should use WithPreviewProps instead

File: packages/admin/cms-admin/src/react-split.d.ts (hunk 7 above)
Comment (manuelblum) [reply packages/admin/cms-admin/src/react-split.d.ts:1]: I don't think we need this. `react-split` has its own types!

Screenshot 2024-12-05 at 07 25 15

File: knip.json (hunk 2 above)
Comment (manuelblum) [reply knip.json:116]: also turned off in start now.

File: packages/admin/admin-color-picker/src/ColorPicker.tsx (hunk 10 above)
Comment (manuelblum) [reply packages/admin/admin-color-picker/src/ColorPicker.tsx]: done

File: packages/admin/admin-date-time/src/DatePickerNavigation.tsx (hunk 11 above)
Comment (manuelblum) [reply packages/admin/admin-date-time/src/DatePickerNavigation.tsx]: done

File: packages/admin/admin-date-time/src/dateTimePicker/DateTimePicker.tsx (hunk 4 above)
Comment (manuelblum) [reply packages/admin/admin-date-time/src/dateTimePicker/DateTimePicker.tsx]: done

File: packages/admin/admin-date-time/src/timePicker/FinalFormTimePicker.tsx (hunk 5 above)
Comment (manuelblum) [reply packages/admin/admin-date-time/src/timePicker/FinalFormTimePicker.tsx]: done

File: packages/admin/admin-date-time/src/timePicker/TimePicker.tsx (hunk 12 above)
Comment (manuelblum) [reply packages/admin/admin-date-time/src/timePicker/TimePicker.tsx]: done.

File: packages/admin/admin-date-time/src/timeRangePicker/TimeRangePicker.tsx (hunk 13 above)
Comment (manuelblum) [reply packages/admin/admin-date-time/src/timeRangePicker/TimeRangePicker.tsx]: done.

File: packages/admin/admin/src/appHeader/fillSpace/AppHeaderFillSpace.tsx (hunk 14 above)
Comment (manuelblum) [reply packages/admin/admin/src/appHeader/fillSpace/AppHeaderFillSpace.tsx]: done.

File: packages/admin/admin/src/table/TableDndOrder.tsx (hunk 17 above)
Comment (manuelblum) [reply packages/admin/admin/src/table/TableDndOrder.tsx]: done.

File: packages/admin/blocks-admin/src/blocks/factories/createListBlock.tsx (hunk 19 above)
Comment (manuelblum) [reply packages/admin/blocks-admin/src/blocks/factories/createListBlock.tsx]: done.

File: packages/admin/blocks-admin/src/blocks/factories/createOptionalBlock.tsx (hunk 21 above)
Comment (manuelblum) [reply packages/admin/blocks-admin/src/blocks/factories/createOptionalBlock.tsx]: done.

File: packages/admin/cms-admin/src/blocks/PixelImageBlock.tsx (hunk 22 above)
Comment (manuelblum) [reply packages/admin/cms-admin/src/blocks/PixelImageBlock.tsx:35]: i am nut sure either.

File: packages/admin/cms-admin/src/generator/future/generator.ts (hunk 23 above)
Comment (manuelblum) [reply packages/admin/cms-admin/src/generator/future/generator.ts:116]: yes it is 

```
export type GeneratorConfig = FormConfig<any> | GridConfig<any> | TabsConfig;
```

File: packages/site/cms-site/src/iframebridge/withPreview.tsx (hunk 24 above)
Comment (manuelblum) [reply packages/site/cms-site/src/iframebridge/withPreview.tsx:15]: ok,... so we keep the change and keep it private.

File: packages/admin/blocks-admin/src/blocks/factories/createOneOfBlock.tsx (hunk 20 above)
Comment (manuelblum) [reply packages/admin/blocks-admin/src/blocks/factories/createOneOfBlock.tsx:52]: done.

File: packages/admin/admin/src/common/ClearInputAdornment.tsx (hunk 15 above)
Comment (manuelblum) [reply packages/admin/admin/src/common/ClearInputAdornment.tsx]: done.

File: packages/admin/admin/src/index.ts (hunk 6 above)
Comment (manuelblum) [reply packages/admin/admin/src/index.ts:34]: i dont know, why should we need a changeset for this? Because the public api got extended?

File: demo/admin/codegen.ts (hunk 1 above)
Comment (manuelblum) [reply demo/admin/codegen.ts:7]: knip is somehow loading all source files and tries to resolve schema.gql. due to knip run's in the root directory it can not find schema.gql file.

File: package.json (hunk 3 above)
Comment (manuelblum) [reply package.json:50]: that's needed because knip has a minimum peer dependency of 5.0.4

image

File: packages/api/cms-api/src/index.ts (hunk 9 above)
Comment (manuelblum) [reply packages/api/cms-api/src/index.ts:62]: @johnnyomair  There are other factory methods like `createOptionalBlock` or `createSeoBlock` (as an example) which export their interfaces, i thought i would make sense.

We can as well make the BlocksBlockInputInterface private.

File: packages/eslint-plugin/package.json (hunk 8 above)
Comment (manuelblum) [reply packages/eslint-plugin/package.json]: i think, they should be dev dependencies.

File: packages/admin/admin/src/index.ts (hunk 6 above)
Comment (johnnyomair) [reply packages/admin/admin/src/index.ts:35]: Yes

File: demo/api/src/news/generated/dto/news.sort.ts [hunk 25]
```diff
@@ -4,7 +4,7 @@ import { SortDirection } from "@comet/cms-api";
 import { Field, InputType, registerEnumType } from "@nestjs/graphql";
 import { IsEnum } from "class-validator";
 
-export enum NewsSortField {
+enum NewsSortField {
```
Comment (johnnyomair) [inline demo/api/src/news/generated/dto/news.sort.ts:7]: @dkarnutsch argued that the sort enum may be used in scaffolded code here: https://github.com/vivid-planet/comet-starter/pull/613. Maybe we should add a knipignore instead?

File: demo/api/src/open-telemetry/metrics.ts [hunk 26]
```diff
@@ -39,19 +39,3 @@ export function getOrCreateHistogram(name: string, options: MetricOptions = {}):
...
-    return getOrCreate(name, options, MetricType.ObservableGauge) as ObservableGauge;
-}
-
-export function getOrCreateObservableCounter(name: string, options: MetricOptions = {}): ObservableCounter {
-    return getOrCreate(name, options, MetricType.ObservableCounter) as ObservableCounter;
-}
-
-export function getOrCreateObservableUpDownCounter(name: string, options: MetricOptions = {}): ObservableUpDownCounter {
-    return getOrCreate(name, options, MetricType.ObservableUpDownCounter) as ObservableUpDownCounter;
-}
```
Comment (johnnyomair) [inline demo/api/src/open-telemetry/metrics.ts:57]: These were recently introduced by @nsams, so maybe we shouldn't remove them?

File: packages/admin/admin-date-time/src/DatePickerNavigation.tsx [hunk 27]
```diff
@@ -107,7 +107,7 @@ export const DatePickerNavigation = (inProps: DatePickerNavigationProps) => {
     );
 };
 
-export type DatePickerNavigationClassKey =
+type DatePickerNavigationClassKey =
```
Comment (johnnyomair) [inline packages/admin/admin-date-time/src/DatePickerNavigation.tsx:110]: The class key should be added to src/index.ts

File: packages/eslint-config/package.json [hunk 28]
```diff
@@ -33,11 +33,13 @@
         "npm-run-all": "^4.1.5"
     },
     "devDependencies": {
+        "@comet/eslint-plugin": "workspace:^7.7.0",
         "eslint": "^8.32.0",
         "prettier": "^2.0.0",
         "typescript": "^4.0.0"
     },
     "peerDependencies": {
+        "@comet/eslint-plugin": "workspace:^7.7.0",
```
Comment (johnnyomair) [inline packages/eslint-config/package.json:42]: It's already a normal dependency. Why add it as a peer dependency? This would be a breaking change.

Comment (manuelblum) [reply packages/eslint-config/package.json:42]: if the user of the eslint-config does not install this as an dependency - eslint is not working correctly. I thought it would be a good idea to add it to the peer Dependencies, so user of the library, at least know that they have to install it.

File: demo/admin/codegen.ts (hunk 1 above)
Comment (manuelblum) [reply demo/admin/codegen.ts:7]: Because we have no monorepo in the starter. In the starter we are executing 3 times knip - in each service (api/admin/site) so the execution directory is different like in this repository.

File: packages/admin/admin/src/index.ts (hunk 6 above)
Comment (manuelblum) [reply packages/admin/admin/src/index.ts:35]: so we are saying that we export more types and more components now - right?

File: demo/api/src/news/generated/dto/news.sort.ts (hunk 25 above)
Comment (manuelblum) [reply demo/api/src/news/generated/dto/news.sort.ts:7]: ok, thats basically like i said -> then i will revert this change, because we do not really know if someone will be using those enums.

File: demo/api/src/open-telemetry/metrics.ts (hunk 26 above)
Comment (manuelblum) [reply demo/api/src/open-telemetry/metrics.ts:57]: ok, but @nsams is not using those function. In my opinion we should remove that dead code.

File: packages/admin/admin-date-time/src/DatePickerNavigation.tsx (hunk 27 above)
Comment (manuelblum) [reply packages/admin/admin-date-time/src/DatePickerNavigation.tsx]: exported

File: packages/eslint-config/package.json (hunk 28 above)
Comment (johnnyomair) [reply packages/eslint-config/package.json:42]: But the plugin is a dependency of the config 🤔

File: packages/admin/admin/src/index.ts (hunk 6 above)
Comment (manuelblum) [reply packages/admin/admin/src/index.ts:35]: added changset for all packages / public api changes.

File: demo/api/src/news/generated/dto/news.sort.ts (hunk 25 above)
Comment (johnnyomair) [reply demo/api/src/news/generated/dto/news.sort.ts:7]: sorry, my bad.

Comment (manuelblum) [reply demo/api/src/news/generated/dto/news.sort.ts:7]: shall we add explicit knip ignore to the enum field? or shall we globally ignore all generated/** files? 

I would prefer to ignore all generated files from linting.

File: .changeset/clean-dots-pull.md [hunk 29]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": patch
+---
+
+Add BlocksBlockInputInterface to public api
```
Comment (johnnyomair) [inline .changeset/clean-dots-pull.md:5]:
```suggestion
Add `BlocksBlockInputInterface` to the public API
```

File: .changeset/four-forks-call.md [hunk 30]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/admin-date-time": patch
+---
+
+Add DatePickerNavigationClassKey, DatePickerNavigationProps, DateTimePickerClassKey, FinalFormTimePickerProps, TimePickerClassKey, TimeRangePickerClassKey to public api
```
Comment (johnnyomair) [inline .changeset/four-forks-call.md:5]:
```suggestion
Add `DatePickerNavigationClassKey`, `DatePickerNavigationProps`, `DateTimePickerClassKey`, `FinalFormTimePickerProps`, `TimePickerClassKey`, and `TimeRangePickerClassKey` to the public API
```

File: .changeset/khaki-zebras-roll.md [hunk 31]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-site": patch
+---
+
+Add AdminMessageType, IAdminContentScopeMessage, IAdminGraphQLApiUrlMessage, IAdminHoverComponentMessage, IAdminShowOnlyVisibleMessage, IFrameHoverComponentMessage, IFrameLocationMessage, IFrameMessage, IFrameMessageType, IFrameOpenLinkMessage, IFrameSelectComponentMessage, IReadyIFrameMessage to public api
```
Comment (johnnyomair) [inline .changeset/khaki-zebras-roll.md:5]:
```suggestion
Add `AdminMessageType`, `IAdminContentScopeMessage`, `IAdminGraphQLApiUrlMessage`, `IAdminHoverComponentMessage`, `IAdminShowOnlyVisibleMessage`, `IFrameHoverComponentMessage`, `IFrameLocationMessage`, `IFrameMessage`, `IFrameMessageType`, `IFrameOpenLinkMessage`, `IFrameSelectComponentMessage`, and `IReadyIFrameMessage` to the public API
```

File: .changeset/lucky-books-thank.md [hunk 32]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/admin-color-picker": patch
+---
+
+Add ColorPickerNoColorPreviewProps to public api
```
Comment (johnnyomair) [inline .changeset/lucky-books-thank.md:5]:
```suggestion
Add `ColorPickerNoColorPreviewProps` to the public API
```

File: .changeset/sharp-apes-visit.md [hunk 33]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/admin": patch
+---
+
+Add AppHeaderFillSpaceProps, ClearInputAdornmentClassKey, ToolbarActionButtonClassKey, ToolbarActionButton, CrudMoreActionsMenuClassKey, GridActionsColDef, GridBaseColDef, GridColDef, GridSingleSelectColDef, TableDndOrderClassKey to public api
```
Comment (johnnyomair) [inline .changeset/sharp-apes-visit.md:5]:
```suggestion
Add `AppHeaderFillSpaceProps`, `ClearInputAdornmentClassKey`, `ToolbarActionButtonClassKey`, `ToolbarActionButton`, `CrudMoreActionsMenuClassKey`, `GridActionsColDef`, `GridBaseColDef`, `GridSingleSelectColDef`, and `TableDndOrderClassKey` to the public API
```

File: .changeset/spicy-bottles-admire.md [hunk 34]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-site": patch
+---
+
+Add AdminMessageType, IAdminContentScopeMessage, IAdminGraphQLApiUrlMessage, IAdminHoverComponentMessage, IAdminShowOnlyVisibleMessage, IFrameHoverComponentMessage, IFrameLocationMessage, IFrameMessage, IFrameMessageType, IFrameOpenLinkMessage, IFrameSelectComponentMessage, IReadyIFrameMessage to public api
```
Comment (johnnyomair) [inline .changeset/spicy-bottles-admire.md:5]:
```suggestion
Add `AdminMessageType`, `IAdminContentScopeMessage`, `IAdminGraphQLApiUrlMessage`, `IAdminHoverComponentMessage`, `IAdminShowOnlyVisibleMessage`, `IFrameHoverComponentMessage`, `IFrameLocationMessage`, `IFrameMessage`, `IFrameMessageType`, `IFrameOpenLinkMessage`, `IFrameSelectComponentMessage`, and `IReadyIFrameMessage` to the public API
```

File: .changeset/wicked-coins-work.md [hunk 35]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/blocks-admin": patch
+---
+
+Add BlocksBlockOutput, ListBlockOutput, OneOfBlockOutput, OneOfBlockPreviewState, OptionalBlockOutput to public api
```
Comment (johnnyomair) [inline .changeset/wicked-coins-work.md:5]:
```suggestion
Add `BlocksBlockOutput`, `ListBlockOutput`, `OneOfBlockOutput`, `OneOfBlockPreviewState`, and `OptionalBlockOutput` to the public API
```

File: demo/api/src/open-telemetry/metrics.ts (hunk 26 above)
Comment (nsams) [reply demo/api/src/open-telemetry/metrics.ts:57]: We will add more metrics in future - that might need those functions. It's a implementation for all available metric types. Please keep it (this is demo). Feel free to comment them out.

File: demo/admin/package.json [hunk 36]
```diff
@@ -3,7 +3,7 @@
     "version": "1.0.0",
     "private": true,
     "scripts": {
-        "admin-generator": "rimraf 'src/*/generated' && comet-admin-generator generate crud-generator-config.ts && comet-admin-generator future-generate",
+        "admin-generator": "rimraf 'src/*/generated' && comet-admin-generator generate crud-generator-config.ts && comet-admin-generator future-generate && pnpm gql:types",
```
Comment (nsams) [inline demo/admin/package.json:6]: is that change a good idea?

File: demo/api/src/news/generated/dto/news.sort.ts (hunk 25 above)
Comment (johnnyomair) [reply demo/api/src/news/generated/dto/news.sort.ts:7]: > I would prefer to ignore all generated files from linting.

Then let's do that. Please ignore the files generated by the CRUD Generators

File: packages/eslint-config/package.json [hunk 37]
```diff
@@ -33,6 +33,7 @@
         "npm-run-all": "^4.1.5"
     },
     "devDependencies": {
+        "@comet/eslint-plugin": "workspace:^7.7.0",
```
Comment (nsams) [inline packages/eslint-config/package.json:36]: this is already a `"dependency"`

File: demo/api/src/open-telemetry/metrics.ts (hunk 26 above)
Comment (manuelblum) [reply demo/api/src/open-telemetry/metrics.ts]: added out commented code back. My heart bleeds 🩸 a little bit.

File: demo/admin/package.json (hunk 36 above)
Comment (manuelblum) [reply demo/admin/package.json:6]: due to the admin generator is not automatically generating types for it's own, and those generated gql types get deleted everytime. I had to run this command over and over again and forgot to generate the gql:types as well.

Maybe its out of scope from this pr, i will revert this, but it's probably worth to add this later on?

Comment (manuelblum) [reply demo/admin/package.json]: removed it.

File: packages/eslint-config/package.json (hunk 37 above)
Comment (manuelblum) [reply packages/eslint-config/package.json]: absolutely correct. thanks for mentioning it. 

removed it.

File: demo/api/src/news/generated/dto/news.sort.ts (hunk 25 above)
Comment (manuelblum) [reply demo/api/src/news/generated/dto/news.sort.ts]: done. ignored generated directories in demo/api and demo/admin

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]
